### PR TITLE
Small fixes part x

### LINKED ISSUE #348: Lazy sequences in routing table cause strange endresultss
## Library Version(s)

`2.0.0-alpha10`

## Problem

Currently, to mount sequential routes, one needs to wrap them into `(apply routes my-routes)`. Would be nice to avoid this and support sequential routes with both `routes` and `context`. Something like:

```clj
(context "/api" []
  (for [path ["/ping" "/pong"]]
    (GET path [] (ok {:path path}))))
```
Currently, that fails with:

```clj
CompilerException java.lang.ClassCastException: clojure.lang.LazySeq cannot be cast to clojure.lang.IFn
```

### PR DESCRIPTION
**Pull request by ikitommi**
* Support `sequential?` routes
* Update deps
* drop `dynamic-context` in favour of `:dynamic` meta (less syntax, more meta)

Fixes #348

### CODE REVIEW

File: src/compojure/api/meta.clj [hunk 1]
```diff
@@ -607,7 +635,8 @@
                           :body body}
                          options)
 
-        static? (not (or dynamic? (route-args? route-arg) (seq lets) (seq letks)))
+        static? (not (or (-> info :public :dynamic boolean)
```
Comment (miikka) [inline src/compojure/api/meta.clj]: Why boolean?

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,47 @@
...
+;      (GET "/ping" [] (ok "pong")))
+;
+; (context "/dynamic" []
+;   :dynamic true
+;   (if (= 0 (random-int 2))
+;      ;; mounted for 50% of requests
+;      (GET "/ping" [] (ok "pong")))
+```
+
+* `routes` and `context` flatten `sequential?` routes:
```
Comment (miikka) [inline CHANGELOG.md]: The example is good, but I'm not sure if flattening is the right word for this. More like: "You can now include sequences of routes in `routes` and `context`" or something like that?

File: src/compojure/api/meta.clj (hunk 1 above)
Comment (ikitommi) [reply src/compojure/api/meta.clj]: It used to be `dynamic?` which indicated a `boolean`. fixed.

File: CHANGELOG.md (hunk 2 above)
Comment (ikitommi) [reply CHANGELOG.md]: valid. fixed.

**Pull request merged**